packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -580,3 +580,71 @@ class _UiKitPlatformView extends LeafRenderObjectWidget {
     renderObject.updateGestureRecognizers(gestureRecognizers);
   }
 }
+
+/// This widget provides the minimal required functionality for a platform view.
+///
+/// The render object associated with this widget is [PlatformViewRenderBox].
+///
+/// This widget exists as a helper for easier implementation of a new platform view.
+/// It is entirely up to the implementer to decide whether to use this helper.
+///
+/// See also:
+/// * [PlatformViewRenderBox]
+class PlatformViewSurface extends LeafRenderObjectWidget {
+
+  /// Construct a `PlatformViewSurface`.
+  ///
+  /// The [id] and the [controller] must not be null.
+  /// By default, this widget will create a [PlatformViewLayer] with rect = size & offset and the same id of
+  /// this [PlatformViewSurface] in the layer tree. If you want to use
+  /// a different type of layer, such as Android,
+  /// or a different configuration, such as supporting different sizes,
+  /// provide a [layerBuilder].
+  const PlatformViewSurface({
+    @required this.id,
+    @required this.controller,
+    PlatformViewRenderBoxLayerBuilder layerBuilder,
+  }) : assert(id != null && id > -1),
+       assert(controller != null),
+      _layerBuilder = layerBuilder;
+
+  /// The id of the platform view that is associate with this [PlatformViewSurface].
+  ///
+  /// The id should always be unique and non-negative.
+  /// In most cases, it should be generated with [PlatformViewsRegistry.getNextPlatformViewId].
+  /// It is used when handling semantics, composition orders and etc for the platform view.
+  final int id;
+
+  /// A controller for a platform view.
+  ///
+  /// Someone who is implementing a new platform view using [PlatformViewSurface] should implement
+  /// a [PlatformViewController] and construct the [PlatformViewSurface] with it.
+  ///
+  /// The controller is responsible for:
+  ///
+  /// * handling clearing focus.
```

Comment:
These still need indendtation, I think something like:

```
The controller is responsible for:

    * handling clearing focus
    * ...
```

Reply:
And why aren't these somehow required on the PlatformViewController interface defined above?